akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -1311,6 +1312,26 @@ final class Flow[In, Out, Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends Gr
   def recover(pf: PartialFunction[Throwable, Out]): javadsl.Flow[In, Out, Mat] =
     new Flow(delegate.recover(pf))
 
+  /**
+   * RecoverTermination allows to transform any kind of termination (success or failure) into either a successful
+   * termination or a failed termination.
+   *
+   * Since the underlying failure signal onError arrives out-of-band, it might jump over existing elements.
+   * This stage can recover the failure signal, but not the skipped elements, which will be dropped.
+   *
+   * Throwing an exception inside `recoverTermination` _will_ be logged on ERROR level automatically.
+   *
+   * '''Emits when''' element is available from the upstream
+   *
+   * '''Backpressures when''' downstream backpressures
+   *
+   * '''Completes when''' upstream completes or upstream failed
+   *
+   * '''Cancels when''' downstream cancels
+   */
+  def recoverTermination[T >: Out](pf: PartialFunction[Try[Done], Try[Done]]): javadsl.Flow[In, Out, Mat] =
```

Comment:
This API will not fly for Java.
PartialFunction is a pain from Java and `Try` is scary since it's Scala